BootloaderCorePkg/Library/FspSupportLib/FspSupportLib.c

```diff
@@ -30,7 +30,13 @@ GetFspNvsDataBuffer (
   UINT32           *Length
   )
 {
-  return GetGuidHobData (HobListPtr, Length, &gFspNonVolatileStorageHobGuid);
+  VOID             *GuidHobData;
+
+  GuidHobData = GetGuidHobData (HobListPtr, Length, &gFspNonVolatileStorageHob2Guid);
+  if (GuidHobData == NULL){
+    GuidHobData = GetGuidHobData (HobListPtr, Length, &gFspNonVolatileStorageHobGuid);
+  }
+  return GuidHobData;
```

Comment:
FSP_NON_VOLATILE_STORAGE_HOB2 has a different format from FSP_NON_VOLATILE_STORAGE_HOB. The HOB now contains a pointer to the actual NVS data and length. Returning the HOB data here will have unexpected results.